File: java/ui/settings/SettingsViewModel.java
package ui.settings;

import android.app.Application;
import android.content.Context;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import data.settings.SettingsRepository;
import data.settings.SettingsRepositoryImpl;

public class SettingsViewModel extends AndroidViewModel {

    private final SettingsRepository repository;

    private final MutableLiveData<Boolean> privateAccount = new MutableLiveData<>();
    private final MutableLiveData<String> themeMode = new MutableLiveData<>();

    public SettingsViewModel(@NonNull Application application) {
        this(application, new SettingsRepositoryImpl(application.getApplicationContext()));
    }

    public SettingsViewModel(@NonNull Application application, @NonNull SettingsRepository repository) {
        super(application);
        this.repository = repository;
        privateAccount.setValue(repository.isPrivateAccount());
        themeMode.setValue(repository.getThemeMode());
    }

    public LiveData<Boolean> getPrivateAccount() { return privateAccount; }
    public LiveData<String> getThemeMode() { return themeMode; }

    public void setPrivateAccount(boolean enabled) {
        repository.setPrivateAccount(enabled);
        privateAccount.setValue(enabled);
    }

    public void setThemeMode(String mode) {
        repository.setThemeMode(mode);
        themeMode.setValue(mode);
    }
}

File: kotlin/ui/settings/SettingsViewModel.kt
package ui.settings

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.viewModelScope
import data.settings.SettingsRepository
import data.settings.SettingsRepositoryImpl
import data.settings.SettingsStore
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.collectLatest
import kotlinx.coroutines.launch

class SettingsViewModel(application: Application) : AndroidViewModel(application) {

    private val repository: SettingsRepository = SettingsRepositoryImpl(SettingsStore(application))

    private val _privateAccount = MutableStateFlow(false)
    val privateAccount: StateFlow<Boolean> = _privateAccount

    private val _themeMode = MutableStateFlow("system")
    val themeMode: StateFlow<String> = _themeMode

    init {
        viewModelScope.launch {
            repository.isPrivateAccount.collectLatest { _privateAccount.value = it }
        }
        viewModelScope.launch {
            repository.themeMode.collectLatest { _themeMode.value = it }
        }
    }

    fun setPrivateAccount(enabled: Boolean) {
        viewModelScope.launch { repository.setPrivateAccount(enabled) }
    }

    fun setThemeMode(mode: String) {
        viewModelScope.launch { repository.setThemeMode(mode) }
    }
}

